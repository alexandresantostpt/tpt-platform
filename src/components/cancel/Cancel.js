import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { Link } from 'react-router-dom'

import { i18n } from '@i18n'

import { classNames } from '@utils/component'

const StyledCancel = styled(Link)`
    color: #3cb1ff;
    margin-top: 1rem;
`

const Cancel = ({ center, link }) => (
    <StyledCancel className={classNames({ center })} to={link}>
        {i18n.t('link.cancel')}
    </StyledCancel>
)

Cancel.defaultProps = {
    center: false
}

Cancel.propTypes = {
    center: PropTypes.bool,
    link: PropTypes.string.isRequired
}

export default Cancel
