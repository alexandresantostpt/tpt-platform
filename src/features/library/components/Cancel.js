import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { i18n } from '@i18n'

const StyledCenter = styled.div`
    text-align: center;
`

const StyledCancel = styled.button`
    background: none;
    border: none;
    color: #3cb1ff;
    margin-top: 1rem;
    &:hover {
        cursor: pointer;
    }
`

const Cancel = ({ onClick }) => (
    <StyledCenter>
        <StyledCancel onClick={onClick}>{i18n.t('link.cancel')}</StyledCancel>
    </StyledCenter>
)

Cancel.propTypes = {
    onClick: PropTypes.func.isRequired
}

export default Cancel
