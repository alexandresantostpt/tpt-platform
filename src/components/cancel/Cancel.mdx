---
menu: Cancel
name: Cancel
route: /cancel
---
import { Playground, PropsTable } from 'docz'

import Cancel from './Cancel'

import '@cfg/scripts'
import '@cfg/style'

# Cancel

<PropsTable of={ Cancel }/>

## Basic Usage

```
<Cancel link={ LINK }/>
```

<Playground>
    <Cancel link="LINK"/>
</Playground>