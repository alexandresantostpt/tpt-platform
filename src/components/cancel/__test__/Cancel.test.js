import React from 'react'
import renderer from 'react-test-renderer'

import Cancel from '@components/cancel/Cancel'

describe('Tests for Cancel component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer.create(<Cancel link={'/teste'} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
