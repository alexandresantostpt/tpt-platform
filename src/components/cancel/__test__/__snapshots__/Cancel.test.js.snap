// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Cancel component Should rendering without crash 1`] = `
<a
  className="Cancel__StyledCancel-sc-1prgtit-0 WzuoR"
  to="/teste"
>
  Cancelar
</a>
`;
